analyze.js

```diff
@@ -104,19 +104,72 @@ async function attachScanFiles(scanGlobs, formData) {
+    core.info(`No target branch was defined. Using default target branch '${defaultBranch}'`)
+    config.targetBranchName = defaultBranch
+    config.baseBranchName = null
+  }
+
+  const baseBranchPresent = checkIfBranchPresent(branches, config.baseBranchName)
+  const targetBranchPresent = checkIfBranchPresent(branches, config.targetBranchName)
+
+  if (targetBranchPresent) {
+    core.info("Using existing SRM branch: " + config.targetBranchName);
+    // Not necessary, base branch is currently ignored in the backend if the target
+    // branch already exists. just setting to null to safeguard in case of future changes
+    config.baseBranchName = null
+  } else {
+    // Base branch is not defined; throw error
+    if (!config.baseBranchName) {
+      throw new Error("A parent branch must be specified when using a new target branch");
+    }
+
+    // Base branch is defined but is not present; cannot create target branch, so throw error
+    if (!baseBranchPresent) {
+      throw new Error("The specified parent branch does not exist: " + config.baseBranchName);
+    }
+
+    // If base branch is defined and present, create the target branch off the base
+    core.info(`Analysis will create a new branch named '${config.targetBranchName}' based on the branch '${config.baseBranchName}'`);
+  }
+}
+
 // most @actions toolkit packages have async methods
 module.exports = async function run() {
   const config = getConfig()
+  const MinimumVersionForBranching = '2022.4.3'
 
-  const client = new CodeDxApiClient(config.serverUrl, config.apiKey, config.caCert)
-  core.info("Checking connection to Code Dx...")
+  const client = new SrmApiClient(config.serverUrl, config.apiKey, config.caCert)
+  core.info("Checking connection to SRM...")
 
-  const codedxVersion = await client.testConnection()
-  core.info("Confirmed - using Code Dx " + codedxVersion)
+  const srmVersion = await client.testConnection()
+  core.info("Confirmed - using SRM " + srmVersion)
 
   core.info("Checking API key permissions...")
   await client.validatePermissions(config.projectId)
-  core.info("Connection to Code Dx server is OK.")
+  core.info("Connection to SRM server is OK.")
+
+  core.info("Validating branch selection...")
+  // First check if the SRM version being used supports branching
+  if (srmVersion.localeCompare(MinimumVersionForBranching, undefined, { numeric: true }) < 0) {
+    core.info(
+        "The connected SRM server with version " + srmVersion + " does not support project branches. " +
+        "The minimum required version is " + MinimumVersionForBranching + ". The target branch and base " +
+        "branch options will be ignored."
+    )
+    // Set both branch configs to null since we will be ignoring them
+    config.baseBranchName = null
+    config.targetBranchName = null
+  } else {
+    const branches = await client.getProjectBranches(config.projectId)
+    await validateBranches(branches, config)
+    core.info("Branch selection is OK.")
+  }
```

Comment:
The branching check should only be done if a `targetBranchName` is actually specified